src/client.rs

```diff
@@ -92,7 +92,13 @@ impl<'a, T: Authenticator + 'a> CDRS<T> {
             let body = start_response.get_body();
             let authenticator = body.get_authenticator().unwrap();
 
-            if authenticator.as_str() == self.authenticator.get_cassandra_name() {
+            if self.authenticator.get_cassandra_name() == "NONE" {
+                let io_err = io::Error::new(
+                    io::ErrorKind::NotFound,
+                    format!("No authenticator was provided "));
+                return Err(error::Error::Io(io_err));
+
+            }else if authenticator.as_str() == self.authenticator.get_cassandra_name().to_string() {
```

Comment:
As for me, I'd like to keep it as it is at least for now.
Enums for me make sense if and only if we have implemented all possible auth strategies which Cassandra may have. Having a separate authenticators + trait `Authenticator` would allow developers to implement whatever authenticator they need in case we don't have it implemented in our trait at the moment.